tidy3d/components/simulation.py

```diff
@@ -89,117 +90,28 @@
 PML_HEIGHT_FOR_0_DIMS = 0.02
 
 
-class Simulation(AbstractSimulation):
+class AbstractYeeGridSimulation(AbstractSimulation, ABC):
+    """
+    Abstract class for a simulation involving electromagnetic fields defined on a Yee grid.
     """
-    Custom implementation of Maxwell’s equations which represents the physical model to be solved using the FDTD
```

Comment:
Hmm from what I can tell it's just organising AbstractYeeGridSimulation and Simulation in the same file. I'm wondering (although maybe a bit painful) if it makes sense to move this to the `base_sim/abstract_yee_grid` maybe thingking that at some point 3.0 it might also make sense to rename our current simulation class `FDTDSimulation` although we can also move this later too! Maybe @tylerflex has some thoughts too

Reply:
I moved all of the fields and methods that weren't FDTD-specific but involve EM fields on Yee grid to `AbstractYeeGridSimulation`. I left duplicates of some methods and fields in `Simulation` when their docstrings had information specific to FDTD simulations.

Reply:
(Not related to this review specifically)

You know I was thinking about this fields/methods code duplicates issue recently. I believe we won't be able to escape it fundamentally in an inheritance-based data scheme (which is good and bad for different reasons), but the alternative which is strongly-typed methods that operate on data container classes have their own set of problems too I guess more on the "user-friendly linking methods to data classes" aspect. This might just be fundamental to having multiple classes have "slightly different variations of the same method" be it in docstrings or implementation. Problems can become more prevalent if we have classmethods using otherclassmethods in a breaking manner that I think we generally try to avoid by writing pure class functions. It is very strongly coupled to the way we choose to represent our data really and have users interact with it, and any approach involves considerable work. There might be a conceptual point of having to draw the line of how exactly we want to interact with our data, and in a way maybe these quirks could be part of it. The real question is the effect this has on scalability and integration with other methods, and how much this affects our development understanding and maintenance of the code. These are not easy questions but I think @tylerflex was thinking about this too in his concept

Reply:
> Hmm from what I can tell it's just organising AbstractYeeGridSimulation and Simulation in the same file. I'm wondering (although maybe a bit painful) if it makes sense to move this to the `base_sim/abstract_yee_grid` maybe thingking that at some point 3.0 it might also make sense to rename our current simulation class `FDTDSimulation` although we can also move this later too! Maybe @tylerflex has some thoughts too

I think this makes sense, but it can be made in a separate PR. We could also go more closely over whether some of the code duplication may be avoided. I'm also down to rename it to `FDTDSimulation`. We may still do something like both `import FDTDSimulation` and `import FDTDSimulation as Simulation` in tidy3d init if we want to keep some backwards compatibility.